### PR TITLE
ref: rework gas usage reporting

### LINKED ISSUE #170: Refactor benchmarking infra
Currently, our benchmarking suite is quite rough and has a bunch of code dependent on `e2e`, we should decouple them.

Benches are also coupled to displaying the results, which should not be the case. Once this coupling is gone, it should make adding the results as a comment on a PR much easier, as mentioned in #169

### PR DESCRIPTION
**Pull request by alexfertel**
Partially completes #170

### CODE REVIEW

File: benches/src/report.rs [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+pub struct Reports(Vec<Report>);
+
+impl Reports {
+    pub fn merge_with(mut self, report: Report) -> Self {
+        self.0.push(report);
+        self
+    }
+}
+
+impl Display for Reports {
```
Comment (qalisander) [inline benches/src/report.rs:29]: That looks cool!

**Review by qalisander — APPROVED** [review]
Cool refactoring. Good job!

File: benches/src/access_control.rs [hunk 2]
```diff
@@ -65,51 +65,17 @@ pub async fn bench() -> eyre::Result<()> {
...
-    );
-    println!(
-        "|{}+--------+--------+----------------------|",
-        "-".repeat(name_width + 2)
-    );
-
-    // Print each row.
-    for (func_name, receipt) in receipts {
+    let mut report = Report::new("AccessControl");
+    for (signature, receipt) in receipts {
```
Comment (bidzyyys) [inline benches/src/access_control.rs:69]: IMHO this loop is repeated in each benchmark; Can we have a common code for it? If not, it is fine to leave as it is.

File: benches/src/report.rs (hunk 1 above)
Comment (bidzyyys) [inline benches/src/report.rs:29]: I like the attitude 🚀

**Review by bidzyyys — APPROVED** [review]
LGTM 🚀

File: benches/src/access_control.rs (hunk 2 above)
Comment (alexfertel) [reply benches/src/access_control.rs:69]: yes, that's a second refactor!

Comment (bidzyyys) [reply benches/src/access_control.rs:69]: Maybe implementing `From<T>` trait for `Report`? I mean `Report::from(receipts)`

**Comment by alexfertel** [comment]
@qalisander @bidzyyys Since in the end we didn't need this today, I'll complete the refactor by tomorrow instead.

File: benches/src/access_control.rs (hunk 2 above)
Comment (alexfertel) [reply benches/src/access_control.rs:69]: We can't cause we need the contract name.

**Review by bidzyyys — APPROVED** [review]
Looks awesome 🚀

File: benches/src/access_control.rs (hunk 2 above)
Comment (bidzyyys) [reply benches/src/access_control.rs]: Missed that one.

**Review by qalisander — APPROVED** [review]
Looks good!

**Pull request merged**